extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/ItemRequiresScrutinizer.java

```diff
@@ -0,0 +1,126 @@
+package org.openrefine.wikidata.qa.scrutinizers;
+
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.updates.ItemUpdate;
+import org.wikidata.wdtk.datamodel.interfaces.PropertyIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.Snak;
+import org.wikidata.wdtk.datamodel.interfaces.SnakGroup;
+import org.wikidata.wdtk.datamodel.interfaces.Statement;
+import org.wikidata.wdtk.datamodel.interfaces.Value;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ItemRequiresScrutinizer extends EditScrutinizer {
+
+    public static final String newItem_type = "item-requires-certain-other-statement";
+    public static final String existingItem_type = "item-should-have-certain-other-statement";
+    public static String ITEM_REQUIRES_CONSTRAINT_QID = "Q21503247";
+    public static String ITEM_REQUIRES_PROPERTY_PID = "P2306";
+    public static String ITEM_OF_PROPERTY_CONSTRAINT_PID = "P2305";
+
+    class ItemRequiresConstraint {
+        final PropertyIdValue itemRequiresPid;
+        final List<Value> itemList;
+
+        ItemRequiresConstraint(Statement statement) {
+            List<SnakGroup> specs = statement.getClaim().getQualifiers();
+            PropertyIdValue pid = null;
+            this.itemList = new ArrayList<>();
+            for(SnakGroup group : specs) {
+                for (Snak snak : group.getSnaks()) {
+                    if (group.getProperty().getId().equals(ITEM_REQUIRES_PROPERTY_PID)){
+                        pid = (PropertyIdValue) snak.getValue();
+                    }
+                    if (group.getProperty().getId().equals(ITEM_OF_PROPERTY_CONSTRAINT_PID)){
+                        this.itemList.add(snak.getValue());
+                    }
+                }
+            }
+            this.itemRequiresPid = pid;
+        }
+    }
+
+    @Override
+    public void scrutinize(ItemUpdate update) {
+        Map<PropertyIdValue, Set<Value>> propertyIdValueValueMap = new HashMap<>();
+        for (Statement statement : update.getAddedStatements()) {
+            PropertyIdValue pid = statement.getClaim().getMainSnak().getPropertyId();
+            Value value = statement.getClaim().getMainSnak().getValue();
+            Set<Value> values;
+            if (value != null) {
+                if (propertyIdValueValueMap.containsKey(pid)) {
+                    values = propertyIdValueValueMap.get(pid);
+                } else {
+                    values = new HashSet<>();
+                }
+                values.add(value);
+                propertyIdValueValueMap.put(pid, values);
+            }
+        }
+
+        if (update.isNew()) {
```

Comment:
Isn't it possible to simplify this code by pulling the `if (update.isNew())` inside, just when the `QAWarning` is generated? I think this only affects the type of warning you emit, so that should get rid of a lot of duplicated code.